### PR TITLE
#162199452 Trending articles

### PR DESCRIPTION
**Pull request by wiztemple**
#### What does this PR do?
- Display top 6 trending articles

#### Description of tasks to be completed
- create cards for displaying trending articles
- create a class component for displaying it
- write the scss for aesthetics
- write test for the component

#### How can this be manually tested?
- on the terminal run git clone https://github.com/andela/haven-ah-frontend.git.
- run git checkout ft-design-trending-article-162199452
- run npm install to install the npm packages.
- run npm test to run the unit and integration tests.

#### What are the relevant Pivotal Tracker stories?
- _[#162199452]()_

#### Screenshots(if appropriate)
screenshot 2018-12-03 at 4 49 54 pm

### CODE REVIEW

**Comment by ucheg6** [comment]
Great Job @wiztemple 👍

File: public/styles/App.scss [hunk 1]
```diff
@@ -1,127 +1,13 @@
 @import url('https://fonts.googleapis.com/css?family=Playfair+Display');
-@import url('https://fonts.googleapis.com/css?family=Noto+Serif');
-@import url('https://fonts.googleapis.com/css?family=Sunflower:300');
-
-$heading-font-family: 'Playfair Display', sans-serif;
```
Comment (seyi-adeleke) [inline public/styles/App.scss]: Why did you remove these?

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: src/actionTypes/trendingActionType.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (seyi-adeleke) [inline src/actionTypes/trendingActionType.js]: Remove this excess space

File: src/actions/trendingArticlesAction.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (seyi-adeleke) [inline src/actions/trendingArticlesAction.js]: same here

File: src/components/trendingsection/TrendingCard.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import defaultImage from '../../../public/images/empty.jpeg';
+import timeFormatter from '../../helpers/timeFormatter.jsx';
+
+const { log } = console;
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingCard.jsx]: remove this log

File: src/components/trendingsection/TrendingCard.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+const TrendingCard = (props) => {
+  const {
+    readtime, title, description, images
+  } = props;
+  log('The images: ', images);
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingCard.jsx]: here

File: src/components/trendingsection/TrendingCard.jsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import defaultImage from '../../../public/images/empty.jpeg';
+import timeFormatter from '../../helpers/timeFormatter.jsx';
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingCard.jsx]: should this be .jsx?

File: src/components/trendingsection/TrendingCard.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+TrendingCard.propTypes = {
+  readtime: PropTypes.number,
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingCard.jsx]: add `.isRequired` to all the proptypes,
`PropTypes.number .isRequired`

File: src/components/trendingsection/TrendingSection.jsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { trendingArticlesAction } from '../../actions/trendingArticlesAction';
+import TrendingCard from './TrendingCard.jsx';
+
+import 'bulma/css/bulma.min.css';
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingSection.jsx]: You shouldnt be importing bulma here

File: src/components/trendingsection/TrendingSection.jsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+import './TrendingSection.scss';
+
+class TrendingSection extends PureComponent {
+  componentDidMount() {
+    this.props.trendingArticlesAction();
+  }
+
+  render() {
+    const { articles } = this.props;
+    console.log(articles);
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingSection.jsx]: Remove this console.log

File: src/components/trendingsection/TrendingSection.scss [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    line-height: 160%;
+    color: #777;
+}
+.ah-card__title > h1 {
+    font-weight: 900;
+    line-height: 150%;
+    padding-bottom: 5px;
+}
+.ah-card__bookmark:hover {
+    color: #a0a0a0;
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingSection.scss]: lets use variables for all our colors

File: src/components/trendingsection/TrendingSection.scss [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    background-size: 100%;
+    background-repeat: no-repeat;
+    border-radius: 10px;
+}
+.ah-card__image-container {
+ width: 100%;
+    height: 180px;
+}
+.ah-card__author-name {
+
```
Comment (seyi-adeleke) [inline src/components/trendingsection/TrendingSection.scss]: fix all the indentation issues

File: src/routes.js [hunk 12]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (seyi-adeleke) [inline src/routes.js]: Remove this space

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 13]
```diff
@@ -1,14 +1,13 @@
...
-import Routes from './routes';
-import store from './store';
-
 import 'bulma/css/bulma.min.css';
 import 'font-awesome/css/font-awesome.min.css';
+import Routes from './routes';
+import store from './store';
 import '../public/styles/App.scss';
 import '../public/styles/responsive.scss';
+import '../public/styles/colors.scss';
```
Comment (seyi-adeleke) [inline src/App.jsx]: is this import necessary?

File: public/styles/colors.scss [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+$red-color: red;
+$icon-grey: #BEBEBE;
+$white-color: #fff;
+$white-grey: #eee;
+$facebook-blue: #4267b2;
+$hover-black: rgba(0,0,0,0.2);
+$facebook-blue-hover: rgba(42, 84, 168, 0.8);
+$twitter-color: rgb(29, 161, 242);
+$twitter-color-hover: rgba(29, 161, 242, 0.8);
+$dark-gray: #a0a0a0;
```
Comment (PriscillaSam) [inline public/styles/colors.scss:11]: Great work @wiztemple

Comment (TheoOkafor) [reply public/styles/colors.scss:11]: nice

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**